src/components/screensavermanager.js

```diff
@@ -18,6 +20,42 @@ define(["events", "playbackManager", "pluginManager"], function (events, playbac
         }
     });
 
+    function getScreensaverPlugin() {
+
+        var apiClient = connectionManager.currentApiClient();
+
+        if (apiClient && apiClient.isLoggedIn()) {
+            isLoggedIn = true;
+        }
+        var option = function () {
+            try {
+                return userSettings.get("screensaver", false);
+            } catch (err) {
+                return null;
+            }
+        }();
+
+        var defaultOption = function (isLoggedIn) {
+            if (isLoggedIn) {
+                return "backdropscreensaver";
+            }
+
+            return "logoscreensaver";
+        }(isLoggedIn);
```

Comment:
Can you explain why you wanted to do it so complex instead of
```js
var option;
var defaultOption = isLoggedIn ? "backdropscreensaver" : "logoscreensaver";

try {
    option = userSettings.get("screensaver", false);
} catch (err) {
    option = null;
}
```
?

Reply:
https://github.com/jellyfin/jellyfin-web/pull/592/commits/f28074fb9ed25b6566c0dd58ee8b3ccec96e4adf